### PR TITLE
Fix race condition in testAppWithWorker that can fail occasionally.

### PR DESCRIPTION
**Pull request by chtyim**

### CODE REVIEW

File: cdap-unit-test/src/test/java/co/cask/cdap/test/app/TestFrameworkTestRun.java [hunk 1]
```diff
@@ -603,14 +603,30 @@ public void testAppWithWorker() throws Exception {
     ApplicationManager applicationManager = deployApplication(testSpace, AppWithWorker.class);
     LOG.info("Deployed.");
```
Comment (bdmogal) [inline cdap-unit-test/src/test/java/co/cask/cdap/test/app/TestFrameworkTestRun.java]: should we remove this log?

**Comment by bdmogal** [comment]
LGTM

File: cdap-unit-test/src/test/java/co/cask/cdap/test/app/TestFrameworkTestRun.java (hunk 1 above)
Comment (chtyim) [inline cdap-unit-test/src/test/java/co/cask/cdap/test/app/TestFrameworkTestRun.java]: Not the focus of this PR. Not going to change it now.

**Comment by chtyim** [comment]
Build passed http://builds.cask.co/browse/CDAP-DUT2482-3

**Pull request merged**